### PR TITLE
Add tests for Vote transaction.

### PR DESCRIPTION
**Pull request by MSNTCS**
This PR will be rebased after #483 is merged.

### CODE REVIEW

**Comment by MSNTCS** [comment]
@majecty Please review the last two commits. The rest of the commits are from another [PR](#483 ) and will be rebased after it is approved.

File: voting/src/voting.rs [hunk 1]
```diff
@@ -133,12 +133,10 @@ impl Context {
                 self.storage_mut().set(vote_id.as_ref(), serde_cbor::to_vec(&vote).unwrap());
 
                 let meeting_id = vote_paper.get_general_meeting_id();
-
```
Comment (majecty) [inline voting/src/voting.rs:135]: Not related to this commit.

File: voting/src/voting.rs [hunk 2]
```diff
@@ -133,12 +133,10 @@ impl Context {
...
                 let mut vote_box = self.meeting_mut().get_vote_box(&meeting_id).unwrap();
                 vote_box.drop_in_box(vote.vote_id.clone());
 
                 self.meeting_mut().update_vote_box(&meeting_id, &vote_box).expect("We get the vote box before");
-
```
Comment (majecty) [inline voting/src/voting.rs:140]: Not related to this commit.

File: voting/tests/common/scenarios.rs [hunk 3]
```diff
@@ -127,6 +127,25 @@ fn tx_vote_paper(
     Transaction::new("Create_Vote_Paper".to_owned(), serde_cbor::to_vec(&tx).unwrap())
 }
 
+fn tx_vote(
+    public: &Public,
+    private: &Private,
+    vote_paper_id: H256,
+    choice: VoteChoice,
+    voter_signature: Signature,
```
Comment (majecty) [inline voting/tests/common/scenarios.rs:135]: `public` and `private` is not used.,
```suggestion
    vote_paper_id: H256,
    choice: VoteChoice,
    voter_signature: Signature,
```

File: voting/tests/common/scenarios.rs [hunk 4]
```diff
@@ -241,3 +260,66 @@ pub fn test_create_vote_paper(ctx: &RwLock<Context>) {
...
+    );
+    let vote_paper_execution =
+        ctx.write().tx_owners.get_mut("voting").unwrap().execute_transaction(&create_vote_paper).unwrap();
+
+    let vote_paper_id: H256 = serde_cbor::from_slice(&vote_paper_execution.events.get(0).unwrap().value).unwrap();
+    let signature = ckey::sign("vote".as_ref(), voter1.private());
+    let choice = VoteChoice::Favor;
+    let vote = tx_vote(voter1.public(), voter1.private(), vote_paper_id, choice.clone(), signature);
+
+    ctx.write().tx_owners.get_mut("voting").unwrap().execute_transaction(&vote);
```
Comment (majecty) [inline voting/tests/common/scenarios.rs:313]: Please check whether `execute_transactionnl` succeed or not.

**Review by majecty — CHANGES_REQUESTED** [review]

File: voting/src/voting.rs [hunk 5]
```diff
@@ -186,10 +191,30 @@ impl TxOwner for Context {
...
+                }
+                TxResult::VoteId(id) => {
+                    let event = Event {
+                        key: "Vote".to_string(),
+                        value: serde_cbor::to_vec(&id.id).unwrap(),
+                    };
+                    Ok(TransactionOutcome {
+                        events: vec![event],
+                    })
+                }
```
Comment (majecty) [inline voting/src/voting.rs:215]: Great!

File: voting/src/voting.rs [hunk 6]
```diff
@@ -134,6 +134,7 @@ impl Context {
                 let vote = Vote::new(vote_paper_id, choice, voter_signature);
                 let vote_id = vote.vote_id.clone();
                 vote_paper.set_used_in(vote_id.clone())?;
+                self.storage_mut().set(vote_paper.vote_paper_id.0.as_ref(), serde_cbor::to_vec(&vote_paper).unwrap());
```
Comment (majecty) [inline voting/src/voting.rs:137]:
```suggestion
                self.storage_mut().set(vote_paper.vote_paper_id.as_ref(), serde_cbor::to_vec(&vote_paper).unwrap());
```

**Review by majecty — CHANGES_REQUESTED** [review]

**Review by majecty — APPROVED** [review]

**Pull request merged**